### ISSUE TITLE
Do not trigger IDE1006 name violation if it cannot be fixed

### ISSUE THREAD

**Issue by miloush** [dotnet/roslyn#15464]
**Version Used**: 2.0.0.61104

**Steps to Reproduce**:

```C#
        class MyClass
        {
            public string கி { get; set; }
        }
```

**Expected Behavior**: No violation if the letter does not have a notion of case.

**Actual Behavior**:
![untitled](https://cloud.githubusercontent.com/assets/10546952/20535451/adf67174-b0dc-11e6-85ab-57a611be1224.png)

Similarly with other casing rules.
Maybe instead of char.IsUpper test !char.IsLower? Or skip all casing rules if char.IsUpper==char.IsLower?

**Comment by timopomer** [comment]
Applies strongly to variables named in other languages like hebrew, but honestly shame on anybody that does that

**Comment by Pilchie** [comment]
Already tracked in https://github.com/dotnet/roslyn/issues/15326.
